app-misc/anki/anki-23.12.1.ebuild

```diff
@@ -0,0 +1,955 @@
+			>=dev-python/PyQt6-sip-13.6.0[${PYTHON_USEDEP}]
+			>=dev-python/PyQt6-WebEngine-6.6.0[widgets,${PYTHON_USEDEP}]')
+	)
+	!qt6? (
+		dev-qt/qtgui:5[jpeg,png]
+		dev-qt/qtsvg:5
+		$(python_gen_cond_dep '
+			>=dev-python/PyQt5-5.15.5[gui,network,webchannel,widgets,${PYTHON_USEDEP}]
+			>=dev-python/PyQt5-sip-12.9.0[${PYTHON_USEDEP}]
+			>=dev-python/PyQtWebEngine-5.15.5[${PYTHON_USEDEP}]')
+	)
+"
+RDEPEND="
+	${DEPEND}
+	app-misc/ca-certificates
+	gui? ( ${GUI_RDEPEND} )
+"
+
+BDEPEND="
+	dev-libs/protobuf
+	virtual/pkgconfig
+	>=virtual/rust-1.71.0
+	gui? (
+		${PYTHON_DEPS}
+		app-alternatives/ninja
+		app-arch/unzip
+		net-libs/nodejs
+		$(python_gen_cond_dep 'dev-python/wheel[${PYTHON_USEDEP}]')
+		qt6?	( $(python_gen_cond_dep 'dev-python/PyQt6[${PYTHON_USEDEP}]') )
+		!qt6?	( $(python_gen_cond_dep 'dev-python/PyQt5[${PYTHON_USEDEP}]') )
+	)
+	test? (
+		${RDEPEND}
+		app-text/dvipng
+		app-text/texlive
+		dev-libs/openssl
+		dev-util/cargo-nextest
+		$(python_gen_cond_dep '
+			dev-python/pytest[${PYTHON_USEDEP}]
+			dev-python/mock[${PYTHON_USEDEP}]')
+	)
+"
+
+QA_FLAGS_IGNORED="usr/bin/anki-sync-server
+	usr/lib/python.*/site-packages/anki/_rsbridge.so"
+
+DOC_CONTENTS="Users with add-ons that still rely on Anki's Qt5 GUI can either
+switch to ${CATEGORY}/${PN}[-qt6], or temporarily set the environment variable
+ENABLE_QT5_COMPAT to 1 to have Anki install the previous compatibility code.
+The latter option has additional runtime dependencies. Please take a look
+at this package's optional runtime features for a complete listing.
+\n\nIn an early 2024 update, ENABLE_QT5_COMPAT will be removed, so this is not
+a long-term solution.
+\n\nAnki's user manual is located online at https://docs.ankiweb.net/
+\nAnki's add-on developer manual is located online at
+https://addon-docs.ankiweb.net/
+"
+
+pkg_setup() {
+	export PROTOC_BINARY="${BROOT}"/usr/bin/protoc
+	export ZSTD_SYS_USE_PKG_CONFIG=1
+
+	if use gui; then
+		python-single-r1_pkg_setup
+		export NODE_BINARY="${BROOT}"/usr/bin/node
+		if ! use debug; then
+			if tc-is-lto; then
+				export RELEASE=2
+			else
+				export RELEASE=1
+			fi
+		fi
+	fi
+}
+
+src_prepare() {
+	default
+	rm -r ftl/{core,qt}-repo || die
+	ln -s "${WORKDIR}"/anki-core-i18n-${COMMITS[ftl-core]} ftl/core-repo || die
+	ln -s "${WORKDIR}"/anki-desktop-ftl-${COMMITS[ftl-desktop]} ftl/qt-repo || die
+
+	mkdir out || die
+	echo -e "${COMMITS[anki]:0:8}" > out/buildhash || die
+	if use gui; then
+		mv "${WORKDIR}"/node_modules out || die
+
+		# Some parts of the runner build system expect to be in a git repository
+		mkdir .git || die
+
+		# Creating the pseudo venv early skips pip dependency checks in src_compile.
+		# out/pyenv/bin/python provides the expected output of the pyenv:bin target.
+		mkdir -p out/pyenv/bin || die
+		ln -s "${PYTHON}" out/pyenv/bin/"${EPYTHON}" || die
+		ln -s "${EPYTHON}" out/pyenv/bin/python || die
+		# TODO: ln -s "${BROOT}/usr/bin/protoc-gen-mypy" out/pyenv/bin || die
```

Comment:
Atm, pseudo-venv.patch removes the code that generates stubs for Anki's protobuf files. I figured that might be ok for now, since it's "only" type hints that are missing.
I have an ebuild for mypy-protobuf on my disk but I struggle to set up the tests correctly, so I haven't published it yet.